executor.go

```diff
@@ -52,10 +53,34 @@ func newExecutor(cfg *executorConfig) *executor {
 func (s *executor) run(mainCtx context.Context,
 	statusChan chan<- SwapInfo) error {
 
-	blockEpochChan, blockErrorChan, err :=
-		s.lnd.ChainNotifier.RegisterBlockEpochNtfn(mainCtx)
-	if err != nil {
-		return err
+	var (
+		err            error
+		blockEpochChan <-chan int32
+		blockErrorChan <-chan error
+	)
+
+	for {
+		blockEpochChan, blockErrorChan, err =
+			s.lnd.ChainNotifier.RegisterBlockEpochNtfn(mainCtx)
+		if err != nil {
+			if strings.Contains(err.Error(),
+				"in the process of starting") {
+
+				// Give chain notifier some time to start and
+				// try to re-attempt block epoch subscription.
+				select {
+				case <-time.After(50 * time.Millisecond):
+					continue
+
+				case <-mainCtx.Done():
+					return err
+				}
+			}
+
+			return err
+		}
+
+		break
```

Comment:
Do we need to break? Couldn't we just return err here? Either it's nil or it's a non-"process of starting" error.

Reply:
The function continues below. So getting here means we've succeeded with the connection.